lib/colonel/search/elasticsearch_provider.rb

```diff
@@ -183,8 +183,8 @@ def put_mapping!(index_name, type_name, revision_type_name, item_mapping, revisi
           revision_type_name=> revision_mapping
         }
 
-        es_client.indices.put_mapping index: index_name, type: type_name, body: item_body
         es_client.indices.put_mapping index: index_name, type: revision_type_name, body: revision_body
+        es_client.indices.put_mapping index: index_name, type: type_name, body: item_body
```

Comment:
Yep. The problem was entirely that you can't create `_parent` entries after the type they point to; you've got to do it before defining the type.  If you have a look at how `item_body` and `revision_body` are defined, you'll see that `revision_body` refers to `type_name`, so `revision_body` has to be passed before `type_name` is defined.